lib/Parser/Parser.cpp

```diff
@@ -596,6 +606,16 @@ class CustomDialectAsmParser : public DialectAsmParser {
     return success(parser.consumeIf(Token::r_square));
   }
 
+  /// Parses a '*' if present.
+  ParseResult parseOptionalStar() override {
```

Comment:
There is a somewhat loose attempt to keep these sorted alphabetically.

Reply:
Can you clarify this a bit more? I'm seeing a list of open-thingy, close-thingy pairs with non-optional and optional variants intermingled. But Less, Greater, LParen, RParen, ... Ellipsis doesn't seem to be alphabetical to me.

Maybe just put Question before Star?

Thanks.

Reply:
Yes. That's why I said "loose". Ellipsis is definitely in the wrong spot.

In general I keep the parseL and parseR next to each other, same with the optional. I'll clean up the outliers in a followup, thanks for pointing them out!